src/migration/1727578711381-VordrUser.ts

```diff
@@ -0,0 +1,87 @@
+import { MigrationInterface, QueryRunner } from "typeorm";
+
+export class VordrUser1727578711381 implements MigrationInterface {
+  name = 'VordrUser1727578711381'
+
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.query(`CREATE OR REPLACE FUNCTION vordr_update_flags()
+RETURNS TRIGGER AS $$
+BEGIN
+  IF NEW.flags @> '{"vordr": true}' THEN
+    -- If vordr is true, set showOnFeed to false and banned to true in post table
+    UPDATE
+      post
+    SET
+      "showOnFeed" = FALSE,
+      "banned" = TRUE,
+      "flags" = jsonb_set(jsonb_set(jsonb_set(flags,
+        '{showOnFeed}', 'false', true),   -- set showOnFeed to false
+        '{vordr}', 'true', true),         -- set vordr to true
+        '{banned}', 'true', true)         -- set banned to true
+    WHERE
+      "authorId" = NEW.id;
+
+    UPDATE
+      comment
+    SET
+      "flags" = jsonb_set(flags, '{vordr}', 'true', true)
+    WHERE
+      "userId" = NEW.id;
+
+  ELSIF NEW.flags @> '{"vordr": false}' THEN
+    -- If vordr is false, set showOnFeed to true and banned to false in post table
```

Comment:
Could it be that the post was banned outside of vordr and then with this we retro unban?

Reply:
Yeah, so should probably add to the where statement where vordr flag is true. So it would only undo vordr'ed posts

Reply:
Added `AND flags @> '{"vordr": true}'`

Reply:
does `@>` support JSON fields? eg. if flags are `{"deleted":true,"vordr":false}` will it also match?

Reply:
Yeah
> ### jsonb @> jsonb → boolean
>
> > Does the first JSON value contain the second? (See [Section 8.14.3](https://www.postgresql.org/docs/current/datatype-json.html#JSON-CONTAINMENT) for details about containment.)
> >
> > '{"a":1, "b":2}'::jsonb @> '{"b":2}'::jsonb → t

Reply:
I guess alternative statement would be `AND flags->>'vordr' = 'true'`

Reply:
OR `AND (flags->>'vordr')::boolean = true`